modules/fedimint-wallet-client/src/cli.rs

```diff
@@ -31,11 +50,40 @@ pub(crate) async fn handle_cli_command(
     let opts = Opts::parse_from(iter::once(&ffi::OsString::from("wallet")).chain(args.iter()));
 
     let res = match opts {
-        Opts::AwaitDeposit { operation_id, num } => {
-            module
-                .await_num_deposit_by_operation_id(operation_id, num)
-                .await?;
-            serde_json::Value::Null
+        Opts::AwaitDeposit {
+            operation_id,
+            num,
+            addr,
+            tweak_idx,
+        } => {
+            if u32::from(addr.is_some())
+                + u32::from(operation_id.is_some())
+                + u32::from(tweak_idx.is_some())
+                != 1
+            {
+                bail!("One and only one of the selector arguments must be set")
+            }
+            if let Some(tweak_idx) = tweak_idx {
+                module.await_num_deposits(tweak_idx, num).await?;
+            } else if let Some(operation_id) = operation_id {
+                module
+                    .await_num_deposits_by_operation_id(operation_id, num)
+                    .await?;
+            } else if let Some(addr) = addr {
```

Comment:
Wondering if there's a  nice abstraction we could use here. But would likely be a macro or very specific to the types here :/